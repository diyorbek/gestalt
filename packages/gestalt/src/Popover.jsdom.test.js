// @flow strict

import { render } from '@testing-library/react';
import PopoverEducational from './PopoverEducational';
import Text from './Text';

describe('PopoverEducational', () => {
  it('renders correctly', () => {
    const element = document.createElement('div');
    const { container } = render(
      <PopoverEducational
        anchor={element}
        accessibilityLabel="text"
        onDismiss={jest.fn()}
        message="text"
        primaryAction={{ text: 'next', role: 'button' }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly with custom children', () => {
    const element = document.createElement('div');
    const { container } = render(
      <PopoverEducational accessibilityLabel="text" anchor={element} onDismiss={jest.fn()}>
        <Text>Custom children</Text>
      </PopoverEducational>,
    );

    expect(container).toMatchSnapshot();
  });

  it('does not render when the anchor is null', () => {
    const { container } = render(
      <PopoverEducational
        accessibilityLabel="text"
        anchor={null}
        onDismiss={() => {}}
        message="text"
        primaryAction={{ text: 'next', role: 'button' }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
