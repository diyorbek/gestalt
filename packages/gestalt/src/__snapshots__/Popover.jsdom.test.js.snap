// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverEducational does not render when the anchor is null 1`] = `<div />`;

exports[`PopoverEducational renders correctly 1`] = `
<div>
  <div
    class="box"
  >
    <div>
      <span
        aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-floating-ui-inert=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
      <div
        class="container rounding4 contents maxDimensions minDimensions"
        style="position: absolute; left: 0px; top: 0px; visibility: visible; transform: translate(0px, 0px);"
        tabindex="-1"
      >
        <div
          class="blue caret"
          style="top: 100%;"
        >
          <svg
            height="4"
            width="12"
          >
            <path
              d="M12 0c-.694 0-1.36.278-1.847.773L7.625 3.34c-.807.819-2.188.885-3.084.148a2.098 2.098 0 01-.163-.148L1.853.775A2.6 2.6 0 000 0"
              stroke="rgb(0 0 0 / 0.02)"
            />
          </svg>
        </div>
        <div
          aria-label="text"
          class="border blueBg blue rounding4 innerContents maxDimensions minDimensions"
          role="dialog"
          style="max-width: 230px;"
        >
          <div
            class="box paddingX4 paddingY4"
            tabindex="0"
          >
            <div
              class="Flex rowGap3 columnGap3 xsDirectionColumn"
            >
              <div
                class="FlexItem"
              >
                <div
                  class="Text fontSize300 lightText alignStart breakWord fontWeightNormal"
                >
                  text
                </div>
              </div>
              <div
                class="FlexItem flexGrow selfEnd"
              >
                <button
                  class="button hideOutline inline accessibilityOutline parentButton"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="button hideOutline inline accessibilityOutline tapTransition md white enabled childrenDiv"
                  >
                    <div
                      class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
                    >
                      next
                    </div>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-floating-ui-inert=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`PopoverEducational renders correctly with custom children 1`] = `
<div>
  <div
    class="box"
  >
    <div>
      <span
        aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-floating-ui-inert=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
      <div
        class="container rounding4 contents maxDimensions minDimensions"
        style="position: absolute; left: 0px; top: 0px; visibility: visible; transform: translate(0px, 0px);"
        tabindex="0"
      >
        <div
          class="blue caret"
          style="top: 100%;"
        >
          <svg
            height="4"
            width="12"
          >
            <path
              d="M12 0c-.694 0-1.36.278-1.847.773L7.625 3.34c-.807.819-2.188.885-3.084.148a2.098 2.098 0 01-.163-.148L1.853.775A2.6 2.6 0 000 0"
              stroke="rgb(0 0 0 / 0.02)"
            />
          </svg>
        </div>
        <div
          aria-label="text"
          class="border blueBg blue rounding4 innerContents maxDimensions minDimensions"
          role="tooltip"
          style="max-width: 230px;"
        >
          <div
            class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            Custom children
          </div>
        </div>
      </div>
      <span
        aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-floating-ui-inert=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;
